src/shims/foreign_items.rs

```diff
@@ -132,13 +133,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 // The implementation is provided by the function with the `#[panic_handler]` attribute.
                 "panic_impl" => {
                     this.check_panic_supported()?;
+                    check_arg_count(args, 1)?;
```

Comment:
This is likewise not needed, because this is not just a shim -- we are finding and returning proper MIR for this call.